source/agora/common/Deserializer.d

```diff
@@ -119,6 +119,54 @@ unittest
         Amount(100), [1, 2, 3]));
 }
 
+/*******************************************************************************
+
+    Default upper bound for `deserializeLength` and friends
+
+    The value is 0x09D0, which is a bit less than a page on most systems,
+    and allows to have allocations taking a full page, as the GC puts some
+    metadata on each page.
```

Comment:
Do we know how big the metadata is? Is it defined somewhere?

IOW how much lower is this upper bound, is it just a guess?

Reply:
Yep it's just a guess, I'd have to clarify this. I'm banking on 256 bytes on x86_64 (2 pointers, 32 bytes of bitmap, and 16 for alignment / canary).

Reply:
It's alright. We can fine-tune it later.